src/main/kotlin/link/infra/packwiz/installer/UpdateManager.kt

```diff
@@ -97,6 +99,105 @@ class UpdateManager internal constructor(private val opts: Options, val ui: IUse
 			handleCancellation()
 		}
 
+		// MultiMC MC and loader version checker
+		val multimcManifestPath = Paths.get(opts.multimcFolder, "mmc-pack.json").toString();
+		val multimcManifestFile = File(multimcManifestPath)
+
+		if (multimcManifestFile.exists()) {
+			ui.submitProgress(InstallProgress("Loading MultiMC pack file..."))
+
+			val multimcManifest = try {
+				JsonParser.parseReader(multimcManifestFile.reader())
+			} catch (e: JsonIOException) {
+				ui.showErrorAndExit("Cannot read the MultiMC pack file", e)
+			} catch (e: JsonSyntaxException) {
+				ui.showErrorAndExit("Invalid MultiMC pack file", e)
+			}.asJsonObject
+
+			Log.info("Loaded MultiMC config")
+
+			// We only support format 1, if it gets updated in the future we'll have to handle that
+			// There's only version 1 for now tho, so that's good
+			if (multimcManifest["formatVersion"].asInt != 1) {
+				ui.showErrorAndExit("Invalid MultiMC format version")
+			}
+
+			var manifestModified = false
+			val modLoaders = hashMapOf("net.minecraftforge" to "forge", "net.fabricmc.fabric-loader" to "fabric")
+			val modLoadersClasses = modLoaders.entries.associate{(k,v)-> v to k}
+			var modLoaderFound = false
+			var modLoader: String? = null
+			var modLoaderOldVer: String? = null
+			var mcOldVer = "Unknown"
+			val components = multimcManifest["components"].asJsonArray
+			for (componentObj in components) {
+				val component = componentObj.asJsonObject
+
+				val version = component["version"].asString
+				when (component["uid"].asString) {
+					"net.minecraft" -> {
+						mcOldVer = version
+						if (version != pf.versions?.get("minecraft")) {
+							manifestModified = true
+							component.addProperty("version", pf.versions?.get("minecraft"))
+						}
+					}
+				}
+				// If we find any of the modloaders we support, we save it and check the version
+				if (modLoaders.containsKey(component["uid"].asString)) {
+					modLoaderFound = true
+					modLoader = modLoaders.getValue(component["uid"].asString)
+					modLoaderOldVer = version
+					if (version != pf.versions?.get(modLoader)) {
+						manifestModified = true
+						component.addProperty("version", pf.versions?.get(modLoader))
+					}
+				}
+			}
+
+			// If we can't find the mod loader in the MultiMC file, we add it
+			if (!modLoaderFound) {
+				for ((loaderClass, loader) in modLoaders) {
+					if (pf.versions?.containsKey(loader) ?: false) {
+						modLoader = loader
+						break
+					}
+				}
+				// If we can't find it in the modpack pack file, something is wrong, we'll stop here
+				if (modLoader == null) ui.showErrorAndExit("Mod Loader not found in modpack pack file")
```

Comment:
Entirely possible to have a packwiz "modpack" with no loader; maybe someone has a pack to distribute their resource packs or the loader is unsupported (which is fine, just continue anyway)